test/core/end2end/fixtures/http_proxy_fixture.c

```diff
@@ -309,41 +309,23 @@ static void on_server_connect_done(grpc_exec_ctx* exec_ctx, void* arg,
 
 /**
  * Parses the proxy auth header value to check if it matches :-
- * Basic <base64_encoded_user_cred>
+ * Basic <base64_encoded_expected_cred>
  * Returns true if it matches, false otherwise
  */
-static bool proxy_auth_header_matches(grpc_exec_ctx *exec_ctx,
-                                      char *proxy_auth_header_val) {
-  if(proxy_auth_header_val == NULL) {
+static bool proxy_auth_header_matches(grpc_exec_ctx* exec_ctx,
+                                      char* proxy_auth_header_val,
+                                      char* expected_cred) {
+  GPR_ASSERT(proxy_auth_header_val != NULL && expected_cred != NULL);
```

Comment:
Let's split this into two separate asserts, so that if they fire, we'll be able to tell which one is the problem.